test/unit_test/workflows/test_run_config.py

```diff
@@ -52,50 +52,63 @@ def test_config_without_azureml_config(self):
+            (
+                {"exclude_environment_credential": True, "exclude_managed_identity_credential": False},
+                (0, 1, 1, 1, 1, 0),
+            ),
+            ({"exclude_environment_credential": True, "exclude_managed_identity_credential": True}, (0, 0, 1, 1, 1, 0)),
+        ],
+    )
+    def test_config_with_azureml_default_auth_params(self, default_auth_params):
+        """
+        default_auth_params[0] is a dict of the parameters to be passed to DefaultAzureCredential
+
+        default_auth_params[1] is a tuple of the number of times each credential is called.
+        the order is totally same with that in DefaultAzureCredential where the credentials
+        are called sequentially until one of them succeeds:
+            EnvironmentCredential -> ManagedIdentityCredential -> SharedTokenCacheCredential
+            -> AzureCliCredential -> AzurePowerShellCredential -> InteractiveBrowserCredential
+        https://learn.microsoft.com/en-us/python/api/azure-identity/azure.identity.defaultazurecredential?view=azure-python # noqa: E501
+        """
         with open(self.user_script_config_file, "r") as f:
             user_script_config = json.load(f)
 
-        user_script_config["azureml_client"]["default_auth_params"] = None
-        config = RunConfig.parse_obj(user_script_config)
-        config.azureml_client.create_client()
-        mocked_env_credential.assert_called_once()
-        mocked_interactive_browser_credential.assert_not_called()
-
-        user_script_config["azureml_client"]["default_auth_params"] = {
-            "exclude_environment_credential": True,
-            "exclude_managed_identity_credential": True,
-        }
-        config = RunConfig.parse_obj(user_script_config)
-        config.azureml_client.create_client()
-        assert mocked_env_credential.call_count == 1
-        assert mocked_managed_identity_credential.call_count == 1
-        assert mocked_azure_cli_credential.call_count == 2
-
-        user_script_config["azureml_client"]["default_auth_params"] = {
-            "exclude_environment_credential": True,
-            "exclude_managed_identity_credential": False,
-        }
+        user_script_config["azureml_client"]["default_auth_params"] = default_auth_params[0]
         config = RunConfig.parse_obj(user_script_config)
         config.azureml_client.create_client()
-        assert mocked_env_credential.call_count == 1
-        assert mocked_managed_identity_credential.call_count == 2
-        assert mocked_azure_cli_credential.call_count == 3
+        assert (
+            self.mocked_env_credentials.call_count,
+            self.mocked_managed_identity_credentials.call_count,
+            self.mocked_shared_token_cache_credentials.call_count,
+            self.mocked_azure_cli_credentials.call_count,
+            self.mocked_azure_powershell_credentials.call_count,
+            self.mocked_interactive_browser_credentials.call_count,
```

Comment:
I saw none of the test cases test mocked_interactive_browser_credentials. Do we need a test case for this? https://github.com/microsoft/Olive/blob/main/olive/azureml/azureml_client.py#L124

Reply:
Added with another function.